docs/snischuk/scripts/pokemons.js

```diff
@@ -0,0 +1,221 @@
+async function initPokemons() {
+    const body = document.querySelector('body');
+    const FORM_FILTERS_ELEMENT = document.querySelector('[data-form-filters]');
+    const SELECT_CUSTOM_BTN = FORM_FILTERS_ELEMENT.querySelector('[data-select-custom-btn]');
+    const SELECT_CUSTOM_LIST = FORM_FILTERS_ELEMENT.querySelector('[data-select-custom-list]');
+    const TEXT_INPUT_ELEMENT = FORM_FILTERS_ELEMENT.elements['search-query'];
+    const CARDS_CONTAINER_ELEMENT = document.querySelector('[data-cards-container]');
+    const LOADER_CONTAINER_ELEMENT = document.querySelector('[data-loader-container]');
+
+    const selectCustomOptions = Array.from(SELECT_CUSTOM_LIST.querySelectorAll('[data-option-value]'));
+
+    let allCards;
+    let filteredCards = [];
+
+    function getSelectedOptionsValues() {
+        return selectCustomOptions.filter((option) => option.classList.contains('filters__select-option--selected'))
+            .map((option) => option.dataset.optionValue);
+    }
+
+    function toggleSelectedOptionHandler(event) {
+        selectCustomOptions.forEach((option) => {
+            if (event.target !== option) return;
+            option.classList.toggle('filters__select-option--selected');
+        });
+    }
+
+    function hideDropDownOptions(event) {
+        event.stopPropagation();
+        if (event.target.tagName === 'LI') return;
+        SELECT_CUSTOM_LIST.classList.remove('filters__select-list--active');
+        body.removeEventListener('click', hideDropDownOptions);
+    }
+
+    function showDropDownOptions(event) {
+        event.stopPropagation();
+        SELECT_CUSTOM_LIST.classList.add('filters__select-list--active');
+        body.addEventListener('click', hideDropDownOptions);
+    }
+
+    function initCustomSelect() {
+        SELECT_CUSTOM_BTN.addEventListener('click', showDropDownOptions);
+        SELECT_CUSTOM_LIST.addEventListener('click', toggleSelectedOptionHandler);
+    }
+
+    function clearContainer(container) {
+        if (!container) return;
+        container.innerHTML = '';
+    }
+
+    function showLoader() {
+        const LOADER_ELELEMNT = `
+            <div class="pokemons__loader-animation"></div>
+            <p class="pokemons__loader-text">Loading...</p>
+        `;
+        LOADER_CONTAINER_ELEMENT.innerHTML = LOADER_ELELEMNT;
+    }
+
+    function hideLoader() {
+        clearContainer(LOADER_CONTAINER_ELEMENT);
+    }
+
+    function createCardElement(pokemon) {
+        const {
+            name,
+            ThumbnailImage,
+            ThumbnailAltText,
+            height,
+            weight,
+            number,
+            type,
+            weakness,
+        } = pokemon;
+
+        const typesButtons = type
+            .map((subtype) => `<button class="card-pokemon__btn card-pokemon__btn--${subtype.toLowerCase()}" type="button">${subtype.charAt(0).toUpperCase()}${subtype.toLowerCase().slice(1)}</button>`)
```

Comment:
I think this is not the buttons - they don't do anything on click. 
![image](https://github.com/MastersAcademy/fe-course-basic-2023/assets/98031768/a3d87846-b7a8-458d-849b-81060e4b4c9a)

Reply:
Fixed